renovate: update source import paths on Go module major updates

tklauser opened the pull request:
Try to update source import paths on major updates. In some cases, API
changes might be necessary but due to the source changes, the
appropriate code owners will already be pulled in for reviews on the PR.

aanm reviewed (changes requested):
@tklauser I think we should keep the major updates otherwise we will never be notified about those. We have two options:
a) leave the configuration as is and fix the PRs manually when they show up
b) add `gomodUpdateImportPaths` in renovate see https://docs.renovatebot.com/modules/manager/gomod/#additional-information and fix the PRs manually if renovate can't do it for some reason

tklauser committed:
renovate: update source import paths on Go module major updates

Try to update source import paths on major updates. In some cases, API
changes might be necessary but due to the source changes, the
appropriate code owners will already be pulled in for reviews on the PR.

Suggested-by: André Martins <andre@cilium.io>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

gandro commented on .github/renovate.json5:
```diff
@@ -74,6 +74,10 @@
+      "postUpdateOptions": [
+	// update source import paths on major updates
+        "gomodUpdateImportPaths",
+      ]
```
@tklauser @aanm Note, there is a missing `,`, c.f. https://github.com/cilium/cilium/issues/23230

tklauser replied:
Whoopsie, opened #24109 to fix it. Thanks for the report.
